src/hooks/chatSidebar.ts

```diff
@@ -87,11 +86,10 @@ export const useChatSidebarInfo = () => {
 };
 
 export const useChatVisibility = () => {
-  const { width } = useWindowDimensions();
   const dispatch = useDispatch();
   useEffect(() => {
-    if (width > LARGE_SCREEN_WIDTH) {
+    if (window.innerWidth > LARGE_SCREEN_WIDTH) {
       dispatch(setChatSidebarVisibility(true));
     }
-  }, [dispatch, width]);
+  }, [dispatch]);
 };
```

Comment:
In removing the `useWindowDimensions` hook, this code will never be triggered automatically on a window resize. Which means it's only going to be 'properly checked' and fix the rendering whenever this screen ends up being re-rendered by something else. Which most likely is going to lead to display issues.

Reply:
I was trying to install this package `react-use` and had lost a fight with TS:
```
/Applications/projects/sparkle/sparkle/node_modules/react-use/lib/useGetSet.d.ts
TypeScript error in /Applications/projects/sparkle/sparkle/node_modules/react-use/lib/useGetSet.d.ts(3,82):
',' expected.  TS1005
```
Can't find the way to deal with this `width` checking. Open for any tips/advise

Reply:
Yup, I just tried locally: `npm i react-use`, then tried adding the following minimal code to a component to test:

```typescript
```

Which gave this error:

```
Failed to compile
/Users/devalias/dev/sparkle/sparkle/node_modules/react-use/lib/useGetSet.d.ts
TypeScript error in /Users/devalias/dev/sparkle/sparkle/node_modules/react-use/lib/useGetSet.d.ts(3,82):
',' expected.  TS1005

    1 | import { Dispatch } from 'react';
    2 | import { IHookStateInitAction, IHookStateSetAction } from './misc/hookState';
  > 3 | export default function useGetSet<S>(initialState: IHookStateInitAction<S>): [get: () => S, set: Dispatch<IHookStateSetAction<S>>];
      |                                                                                  ^
    4 | 
```

Looking at https://stackoverflow.com/questions/46398916/typescript-error-ts1005-expected-ii/51936515 it seems to suggest that this error tends to come up when TypeScript doesn't understand the syntax being used, which supports my theory that we need to updated to a newer version.

This seems to be confirmed in this closed issue on the repo:

- https://github.com/streamich/react-use/issues/1879
  - > Use TS version 4+

We're currently using `"typescript": "^3.9.3"`:

- https://github.com/sparkletown/sparkle/blob/staging/package.json#L156

I bumped this to the latest version (`4.2.4`) locally to test if it would resolve this issue:

- https://github.com/microsoft/TypeScript/releases/tag/v4.2.4
  - https://devblogs.microsoft.com/typescript/announcing-typescript-3-9/#breaking-changes
  - https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#breaking-changes
  - https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/#breaking-changes
  - https://devblogs.microsoft.com/typescript/announcing-typescript-4-2/#breaking-changes

There were a couple of typing issues this uncovered that needed to be fixed in legacy code, but it fixed the error that was coming up based on `react-use`.

Since upgrading TypeScript is unrelated to this PR, I will handle doing this in a new PR. Shall link it here once i've pushed the changes up.

Reply:
See https://github.com/sparkletown/internal-sparkle-issues/issues/160

Reply:
See https://github.com/sparkletown/sparkle/pull/1321 for adding `react-use` as a dependency

Reply:
Thank you a lot for your help. 
I've used your PR https://github.com/sparkletown/internal-sparkle-issues/issues/160 in my own changes to install `react-use`

Reply:
Sounds good. Just make sure those other 2 PRs get merged before this one does so it keeps everything properly segregated :)